src/main/java/wolox/training/repositories/BookRepository.java

```diff
@@ -9,4 +12,9 @@
 	Optional<Book> findByAuthor(String author);
 
 	Optional<Book> findByIsbn(String author);
+
+	@Query("SELECT b FROM Book b WHERE (:publisher is null or b.publisher = :publisher) and (:genre is null or b"
+			+ ".genre = :genre) and (:year is null or b.year = :year)")
+	List<Book> findByPublisherAndGenreAndYear(@Param(value = "publisher") String publisher,
```

Comment:
Why do you need this `@Param` annotation that is from Feign? You are not using Feign here.